url4c41254592_topskychen_AuthTopk_tgz-pJ8-quickhull3d_QuickHull3DTestJ8/src/utility/Compare/genDataOfPointForHadoop.java

```diff
@@ -0,0 +1,77 @@
+package utility.Compare;
+
+import java.io.BufferedOutputStream;
+import java.io.File;
+import java.io.FileNotFoundException;
+import java.io.FileOutputStream;
+import java.io.FileReader;
+import java.io.IOException;
+import java.io.LineNumberReader;
+import java.io.PrintWriter;
+import java.util.ArrayList;
+import java.util.Scanner;
+
+
+import mesh.Delaunay;
+
+import utility.security.Point;
+import jdbm.PrimaryTreeMap;
+import jdbm.RecordManager;
+import jdbm.RecordManagerFactory;
+/*
+ * 
+ * @author qchen
+ * */
+public class genDataOfPointForHadoop {
+	public static boolean DEBUG = true;
+	public static boolean ISCAR = false;
+	public static long SCALE = 100000;
+	public static String filename = "database/PointsData.tmp.dat";
+	
+	public static void main(String[] args) throws IOException{
+		ArrayList<float[]> points = new ArrayList<float[]>();	
+		ArrayList<Integer> idmap = new ArrayList<Integer>();
+		PrintWriter pw = new PrintWriter(new BufferedOutputStream(new FileOutputStream(new File(filename))));
+		try {
+			String line = null;
+			LineNumberReader lr = new LineNumberReader(new FileReader("source/NE.dat"));
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

**Problem**
This line of code might contain a resource leak. Resource leaks can cause your system to slow down or crash.

**Fix**
Consider closing the following resource: *lr*. Execution paths that do not contain closure statements were detected. Ensure *lr* is closed in a try-finally block or declared in a try-with-resources block.

**More info**
[View resource management guidelines at oracle.com](https://www.oracle.com/technetwork/articles/java/trywithresources-401775.html) (external link).